File: discover_more_python/manipulate_the_data.py
from urllib2 import Request, urlopen
from json import loads
from sys import exit

request_headers = {
        'User-Agent': 'Holberton_School',
        'Authorization': 'token 6a54def2525aa32b003337b31487e321d6a2bb59'
    }
url = 'https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc'

request = Request(url, headers=request_headers)
contents = urlopen(request).read()
json_data = loads(contents)

"""Handling the errors. In case of some failure a friendly and clean message is displayed."""
try:
    for name in json_data['items']:
        print name['full_name']
except Exception as error:
    print "Something went wrong!"
    print error
    exit(0)

File: discover_perl/manipulate_the_data.pl
#!/usr/bin/perl

=pod
* ===---Description------------------------------------------------------------===
*   Github Perl most popular Projects, based on Rudy Rigoy projects.
*   Using a Perl script to get the 30 most popular Perl Github repositories
*
*   by Alexandro de Oliveira, for Holberton School
* ===--------------------------------------------------------------------------===
=cut

use strict;
use warnings;
use LWP::UserAgent;
use Data::Dump "pp";            # Used to print and analize complex objects.
#use JSON::Parse ':all';   # To parse JSON data.
use JSON;

my $url = 'https://api.github.com/search/repositories?q=language:perl&sort=stars&order=desc';

my $headers = HTTP::Headers->new(
'User-Agent'=> 'Holberton_School',
'Authorization' => 'token 6a54def2525aa32b003337b31487e321d6a2bb59',
);

my $ua = LWP::UserAgent->new(ssl_opts => { verify_hostname => 0}); # bypass SSL authentication checks.

my $req = HTTP::Request->new(GET => $url, $headers);
# Print the request object:
#print pp($req);

# Requesting:
my $res = $ua->request($req);
# print pp($res);
die "Error getting $url\n", $res->status_line
unless $res->is_success( );

my $myjson = decode_json($res->content());
foreach my $item ( @{$myjson -> {items}})
{
    print $item -> {full_name} . "\n";
}
